Polyfills/Console/Source/Console.cpp

```diff
@@ -34,16 +38,136 @@ namespace
+        size_t size = static_cast<size_t>(size_s);
+        // create a string to hold that size (-1 as we don't want the null terminator)
+        std::string buf(size - 1, '0');
+        // write to there
+        snprintf(buf.data(), size, formatString.c_str(), value);
+        return buf;
+    }
+
     void InvokeCallback(Babylon::Polyfills::Console::CallbackT callback, const Napi::CallbackInfo& info, Babylon::Polyfills::Console::LogLevel logLevel)
     {
-        std::stringstream ss{};
-        for (size_t i = 0; i < info.Length(); i++)
+        std::ostringstream ss{};
+        if (info.Length() > 0)
         {
-            if (i > 0)
+            std::string firstArg = info[0].ToString();
+            size_t currArgIndex = 1;
+
+            // Split the first string into parts limited by substitution characters
+            size_t start = 0;
+            bool matching = false;
+            std::size_t j;
+            for (j = 0; j < firstArg.size(); j++)
+            {
+                const char currChar = firstArg.at(j);
+                if (!matching && currChar == '%')
+                {
+                    matching = true;
+                    if (j > 0)
+                    {
+                        std::string_view prefix(firstArg.data() + start, j - start);
+                        ss << prefix;
+                    }
+                    start = j;
+                }
+                else if (matching && (currChar == 's' || currChar == 'o' || currChar == 'O' || currChar == 'd' || currChar == 'i' || currChar == 'f'))
+                {
+                    matching = false;
+                    std::string_view prefix(firstArg.data() + start, j - start + 1);
+                    // This is a substitution argument, so if we have any remaining arguments to substitute, we should.
+                    // Try to use the next arg for subsitution
+                    if (currArgIndex < info.Length())
+                    {
+                        Napi::Value currArg = info[currArgIndex];
+                        // Check the type of the sub string
+                        if (EndsWith(prefix, 'f'))
+                        {
+                            double d = currArg.ToNumber().DoubleValue();
+                            if (std::isnan(d))
+                            {
+                                ss << "NaN";
+                            }
+                            else
+                            {
+                                // I had to copy the string view to a string here, because the data pointer of the string view
+                                // returns a view of the entire character sequence, which caused an issue on the following
+                                // step when formatting.
+                                std::string copiedSubString(prefix);
+                                std::string formatted = FormatValue(copiedSubString, d);
+                                ss << formatted;
```

Comment:
I'm guessing this is because you need a null-terminated string, which makes sense, but we can probably safely limit the format string to a certain size and allocate the string on the stack instead of the heap.